core/src/components/action-sheet/action-sheet.tsx

```diff
@@ -366,7 +367,7 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
         onIonActionSheetWillDismiss={this.dispatchCancelHandler}
         onIonBackdropTap={this.onBackdropTap}
       >
-        <ion-backdrop tappable={this.backdropDismiss} />
+        <ion-backdrop ref={(el) => (this.backdropEl = el)} tappable={this.backdropDismiss} />
```

Comment:
Looking at these changes you might ask "why?" or "where is this used?". 

The current overlay utils for [`present`](https://github.com/ionic-team/ionic-framework/blob/8e88269d8ee173cc89e578a9db79e0013b4669a4/core/src/utils/overlays.ts#L462-L468) and [`dismiss`](https://github.com/ionic-team/ionic-framework/blob/8e88269d8ee173cc89e578a9db79e0013b4669a4/core/src/utils/overlays.ts#L549C1-L557C2) accept an object that configures the util implementation. In our existing usage, [we pass the entire class instance (`this`) to the function](https://github.com/ionic-team/ionic-framework/blob/8e88269d8ee173cc89e578a9db79e0013b4669a4/core/src/components/action-sheet/action-sheet.tsx#L207). By adding the new `backdropEl` property on the class and assigning the reference in the `render()` function, the util can access it from the implementation without additional configuration. 

This logic applies to all overlays.

Reply:
Out of curiosity, did we consider adding the class directly in the overlay.ts file? The `dismiss` function already checks to see if this is the last overlay we are dismissing. As possible implementation could be:

```js
import { BACKDROP_NO_SCROLL } from '@utils/gesture/gesture-controller';
const present = (...) => {
  document.body.classList.add(BACKDROP_NO_SCROLL);
  ...
}

const dismiss = (...) => {
  if (doc !== undefined && getPresentedOverlays(doc).length === 1) {
    setRootAriaHidden(false);
    document.body.classList.remove(BACKDROP_NO_SCROLL);
  }
}
```

The benefit of this is we don't need to worry about getting and clearing references to the backdrop element by grabbing inside the component instance.

I guess part of this is I don't totally understand why we need to create a new blocker for every overlay presented. I would have expected we'd only block scroll once.

Reply:
Yeah it looks like we could simplify and avoid getting the backdrop element reference by always blocking/unblocking the scroll as part of the overlay implementation. 

Originally I had thought that the behavior was dependent on the overlay having a backdrop, but they all technically render an `ion-backdrop`, it may just not be visible to the user. 

For the scroll blocking implementation we could either reflect the class ourselves like your code or use the `BlockerDelegate` (which does the same thing, but with an API). The only thing inside Ionic Framework that uses this, is this implementation. I'd be open to killing this API in favor of reflecting the class directly - if you agree.

Reply:
I'm fine using the class directly, but lets hold off on killing it entirely because I believe the menu still uses it (even though it doesn't work on iOS, it still works on Android I believe)

Reply:
Actually looks like it's part of the gesture public API, so we'll need to keep it regardless: https://github.com/ionic-team/ionic-framework/blob/89698b338fb05cde427c98720c238d2365abdaa7/core/src/utils/gesture/index.ts#L308